### PR TITLE
Overhaul "What to make public" DI section

### PR DESCRIPTION
**Pull request by mbrodala**
Move the most important info top the top, reduce content duplication, simplify wording and content.

### CODE REVIEW

**Comment by linawolf** [comment]
It would be great if you could review, @sbuerk

File: Documentation/ApiOverview/DependencyInjection/Index.rst [hunk 1]
```diff
@@ -531,29 +531,16 @@ that uses :php:`GeneralUtility::makeInstance()` and :ref:`Extbase controllers
...
-Direct access includes instantiation via :php:`GeneralUtility::makeInstance()`
-with constructor arguments.
-
-This means every class that is directly retrieved using
-:php:`GeneralUtility::makeInstance()` *and* requires dependency injection
-**must** be marked as public. Any other class which requires dependency injection
-and is retrieved by dependency injection itself can be private. Instances of
-:php:`\TYPO3\CMS\Core\SingletonInterface` and Extbase controllers are
-automatically marked as public because they are retrieved using
+Instances of :php:`\TYPO3\CMS\Core\SingletonInterface` and Extbase controllers
```
Comment (froemken) [inline Documentation/ApiOverview/DependencyInjection/Index.rst:542]: I have read this sentence three times and it looks like instanciating classes with GU::makeInstance will "automatically" mark classes as public. Sure, thats wrong. I would prefer to write that in more detail.

Comment (mbrodala) [reply Documentation/ApiOverview/DependencyInjection/Index.rst:542]: lol, interesting that you actually found a 2nd way to read this.

You: fetch via `GU::mI` -> `public: true`
Actual: `public: true` to allow `GU::mI`.

I'll have a look. ;-)

Comment (mbrodala) [reply Documentation/ApiOverview/DependencyInjection/Index.rst:542]: Rephrased now, can you have a look/read?

**Review by froemken — APPROVED** [review]
It's better now ;-)

**Pull request merged**